test/benchmarks/src/main/scala/scala/collection/mutable/OpenHashMapBenchmark.scala

```diff
@@ -0,0 +1,204 @@
+      maps = new Array(n)
+    }
+
+    @Setup(Level.Iteration)
+    def iterationSetup {
+      operations = 0
+    }
+
+    @Setup(Level.Invocation)
+    def setup(params: IterationParams) {
+      for (i <- 0 until maps.length) maps(i) = new OpenHashMap[Int,Int](size)
+
+      if (params.getType == IterationType.MEASUREMENT) {
+        operations += mapEntries
+        System.gc()  // clean up after last invocation
+      }
+    }
+
+    @TearDown(Level.Iteration)
+    def iterationTeardown(params: IterationParams) {
+      if (params.getType == IterationType.MEASUREMENT) {
+        // limit to smaller cases to avoid OOM
+        memory = if (mapEntries <= 1000000) GraphLayout.parseInstance(maps(0), maps.tail).totalSize else 0
+      }
+    }
+  }
+
+  /** State container for the `get()` bulk calling tests.
+    * 
+    * Provides a thread-scoped map of the expected size.
+    * Performs a GC after loading the map.
+    */
+  @State(Scope.Thread)
+  class BulkGetState {
+    val map = new OpenHashMap[Int,Int].empty
+
+    /** Load the map with keys from `1` to `size`. */
+    @Setup
+    def setup(params: BenchmarkParams) {
+      val size = params.getParam("size").toInt
+      put_Int(map, 1, size)
+      System.gc()
+    }
+  }
+
+  /** State container for the `get()` bulk calling tests with deleted entries.
+    * 
+    * Provides a thread-scoped map of the expected size, from which entries have been removed.
+    * Performs a GC after loading the map.
+    */
+  @State(Scope.Thread)
+  class BulkRemovedGetState {
+    val map = new OpenHashMap[Int,Int].empty
+
+    /** Load the map with keys from `1` to `size`, removing half of them. */
+    @Setup
+    def setup(params: BenchmarkParams) {
+      val size = params.getParam("size").toInt
+      put_remove_Int(map, size)
+      System.gc()
+    }
+  }
+
+  /** Put elements into the given map. */
+  private def put_Int(map: OpenHashMap[Int,Int], from: Int, to: Int) {
+    var i = from
+    while (i <= to) {  // using a `for` expression instead adds significant overhead
+      map.put(i, i)
+      i += 1
+    }
+  }
+
+  /** Put elements into the given map, removing half of them as they're added.
+    * 
+    * @param size number of entries to leave in the map on return
+    */
+  def put_remove_Int(map: OpenHashMap[Int,Int], size: Int) {
+    val blocks = 50  // should be a factor of `size`
+    val totalPuts = 2 * size  // add twice as many, because we remove half of them
+    val blockSize: Int = totalPuts / blocks
+    var base = 0
+    while (base < totalPuts) {
+      put_Int(map, base + 1, base + blockSize)
+
+      // remove every other entry
+      var i = base + 1
+      while (i <= base + blockSize) {
+        map.remove(i)
+        i += 2
+      }
+
+      base += blockSize
+    }
+  }
+
+  /** Get elements from the given map. */
+  def get_Int(map: OpenHashMap[Int,Int], size: Int, bh: Blackhole) {
+    var i = 1
+    while (i <= size) {
+      bh.consume(map.get(i).getOrElse(0))
```

Comment:
Or perhaps, put the `Option[Int]` into the blackhole, rather than calling `Option#getOrElse`. Or is the intention to test the combined performance of `OHM#get` / `Option#getOrElse`?

It is also worth thinking about what assumptions these benchmarks make about the scalac compiler options used to compile the tests themselves. Under `-Yopt:l:classpath`, `Option#getOrElse` is inlined 

```
% ~/scala/2.12/bin/scala -Yopt:l:classpath
Welcome to Scala 2.12.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_71).
Type in expressions for evaluation. Or try :help.

scala> class OptInline { def foo(a: Option[Int]) = a.getOrElse(0) }
defined class OptInline

scala> :javap -c OptInline#foo
  public int foo(scala.Option<java.lang.Object>);
    Code:
       0: aload_1
       1: ifnonnull     6
       4: aconst_null
       5: athrow
       6: aload_1
       7: invokevirtual #27                 // Method scala/Option.isEmpty:()Z
      10: ifeq          23
      13: invokestatic  #31                 // Method $line4$$read$$iw$$iw$OptInline$$$anonfun$1:()I
      16: invokestatic  #37                 // Method scala/runtime/BoxesRunTime.boxToInteger:(I)Ljava/lang/Integer;
      19: astore_2
      20: goto          28
      23: aload_1
      24: invokevirtual #41                 // Method scala/Option.get:()Ljava/lang/Object;
      27: astore_2
      28: aload_2
      29: invokestatic  #45                 // Method scala/runtime/BoxesRunTime.unboxToInt:(Ljava/lang/Object;)I
      32: ireturn
```

```
 ~/scala/2.12/bin/scala
Welcome to Scala 2.12.0-M4 (Java HotSpot(TM) 64-Bit Server VM, Java 1.8.0_71).
Type in expressions for evaluation. Or try :help.

scala> class OptInline { def foo(a: Option[Int]) = a.getOrElse(0) }
defined class OptInline

scala> :javap -c OptInline#foo
  public int foo(scala.Option<java.lang.Object>);
    Code:
       0: aload_1
       1: invokedynamic #43,  0             // InvokeDynamic #0:apply$mcI$sp:()Lscala/runtime/java8/JFunction0$mcI$sp;
       6: invokevirtual #49                 // Method scala/Option.getOrElse:(Lscala/Function0;)Ljava/lang/Object;
       9: invokestatic  #55                 // Method scala/runtime/BoxesRunTime.unboxToInt:(Ljava/lang/Object;)I
      12: ireturn
```

As a starting point, I would suggest adding `-Yopt:l:classpath` to the `scalacOptions` for this project. Perhaps we'd like to parameterize this down the track, or maybe have separate sub projects with different optimization levels so we could have fine grained control over the way that benchmark code is compiled.